crates/forge-runner/src/running.rs

```diff
@@ -275,19 +279,34 @@ pub fn run_test_case(
         None
     };
 
-    let run_result = runner.run_function(
+    let mut vm = VirtualMachine::new(true);
+    let run_result = runner.run_function_with_vm(
         func,
+        &mut vm,
         &mut test_execution_syscall_handler,
         hints_dict,
         instructions,
         builtins,
     );
 
+    extend_execution_resources(&mut test_execution_syscall_handler);
+
+    let gas = gas_from_execution_resources(
```

Comment:
It doesn't take into account gas used by `deploy`, it can be missleading. On the other hand I have no idea how to implement it nicely, we would prob have to add additional field to `TestExecutionSyscallHandler` :/ Maybe we should not calculate gas used by deploy at all?

Reply:
I was wrong, aprooving